lib/livebook_web/live/output/input_component.ex

```diff
@@ -7,10 +7,34 @@ defmodule LivebookWeb.Output.InputComponent do
   end
 
   @impl true
-  def update(%{event: :change, value: value}, socket) do
+  def update(%{event: :change, value: value} = assigns, socket) do
     {:ok, handle_change(socket, value)}
   end
 
+  def update(%{attrs: %{type: :datetime}} = assigns, socket) do
+    value = assigns.input_values[assigns.attrs.id]
+
+    socket =
+      socket
+      |> assign(assigns)
+      |> assign(value: value)
+      |> push_event("datetime_input_change:#{assigns.id}", %{datetime: value})
+
+    {:ok, socket}
+  end
```

Comment:
We don't need to push this as events, we can just read `data-utc-value` on the client on `mounted()` and `updated()`, convert into local value and set `input.value` :)